src/DynamicLogger.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace yii\psr;
+
+use Psr\Log\LoggerInterface;
+use Psr\Log\LoggerTrait;
+use WeakReference;
+
+/**
+ * Implements a PSR logger that routes messages to the current Yii Logger.
+ */
+final class DynamicLogger implements LoggerInterface
+{
+    use LoggerTrait;
+    private ?Logger $logger = null;
+    /**
+     * Note that using a weak reference here is merely a good practice.
+     * In reality the ->get() will never return null since the underlying logger doesn't use a weak reference.
```

Comment:
Let's get rid of it if it's not actually used.

Reply:
Fixed in https://github.com/yiisoft/yii2-psr-log-source/pull/8/commits/9eefba76f0b4f2cc78ccba76fef15c74e82b61bd